Created endpoints, made module file structure

s-ieremenko opened the pull request:
Video
https://user-images.githubusercontent.com/76453629/126908088-d00fbf65-c073-4239-9670-03a59a9562d8.mp4

s-ieremenko committed:
created endpoints, made module file structure

s-ieremenko committed:
removed unecessary folders

s-ieremenko committed:
used shorthand syntax for Op.in, removed request as an incoming parameter in services

CatDidenko commented on src/User/user.service.ts:
```diff
@@ -0,0 +1,41 @@
+  roleUuid: string
+): Promise<void> => {
+  let user: User = await User.findByPk(userUuid);
+  let role: Role = await Role.findByPk(roleUuid);
```
Why let? Change to const.

CatDidenko commented on src/common/constants.ts:
```diff
@@ -0,0 +1,24 @@
+  dialect: dbDialect,
+  host: dbHost,
+  models: pathToModels,
+});
```
Move defining sequelize to the separate db file

CatDidenko commented on src/Role/role.service.ts:
```diff
@@ -0,0 +1,51 @@
+): Promise<void> => {
+  const role: Role = await Role.findByPk(roleUuid, { include: Permission });
+  const permission: Permission = await Permission.findByPk(permissionUuid, {
+    include: Role,
```
Is this really needed for permission entity? (include Role)

CatDidenko commented on src/Role/role.service.ts:
```diff
@@ -0,0 +1,51 @@
+  if (!role || !permission) {
+    throw new Error('Incorrect data');
+  }
+  await role.$remove('permissions', role.permissions);
+  role.permissions.push(permission);
+  await role.$add('permissions', role.permissions);
```
It's too complicated. Why you remove all permissions and then add all of them one more time?
You should add permission to the existing array if it not exist in permissions array.

CatDidenko commented on src/Permission/permission.controller.ts:
```diff
@@ -0,0 +1,48 @@
+import express, { Request, Response } from 'express';
+import { badRequest, notFound, ok } from '../common/constants';
+import { PermissionAttributes } from './permission.interfaces';
```
Add blank line after package.json dependencies and before internal project files.

s-ieremenko committed:
replaced set/add with addModel/setModel,
simplified adding permission to role,
moved defining sequelize to the separate file,
removed let, added blank line

s-ieremenko committed:
removed a comment

s-ieremenko committed:
removed [x:string]

s-ieremenko committed:
added method for admin check
